apps/passport-server/src/services/generic-issuance/pipelines/CSVPipeline/makeGenericPODPCD.ts

```diff
@@ -0,0 +1,141 @@
+    // To issue to anyone, `issueToUnmatchedEmail` must be enabled. If it is
+    // not, then some `matchConfig` must be specified.
+    if (issueToUnmatchedEmail !== true && !matchConfig) {
+      throw new Error("No configuration for matching requests to CSV data");
+    }
+
+    if (!issueToUnmatchedEmail) {
+      // If we are matching the user by comparing their email to some value in
+      // the CSV input data:
+      if (matchConfig?.type === "email") {
+        const emailColumnName = matchConfig.inputField;
+        // Check if the column exists
+        if (!(emailColumnName in inputRow)) {
+          throw new Error(
+            `Could not find column "${emailColumnName}" in CSV input data, could not match against user email address`
+          );
+        }
+        // If the value does not match, no POD is issued
+        if (inputRow[emailColumnName] !== requesterEmail) {
+          return undefined;
+        }
+      }
+      // If we are matching the user by comparing their Semaphore ID to some
+      // value in the CSV input data:
+      else if (matchConfig?.type === "semaphoreID") {
+        const semaphoreIDColumnName = matchConfig.inputField;
+        // Check if the column exists
+        if (!(semaphoreIDColumnName in inputRow)) {
+          throw new Error(
+            `Could not find column "${semaphoreIDColumnName}" in CSV input data, could not match against user Semaphore ID`
+          );
+        }
+        // If the value does not match, no POD is issued
+        if (inputRow[semaphoreIDColumnName] !== requesterSemaphoreId) {
+          return undefined;
+        }
+      }
+    }
+
+    const entries: PODEntries = {};
+
+    // Iterate over configured POD entries
+    for (const [name, entryConfig] of Object.entries(output)) {
+      switch (entryConfig.source.type) {
+        // This entry is populated from configuration
+        case "configured":
+          /* @todo non-string values */
+          entries[name] = { type: "string", value: entryConfig.source.value };
+          break;
+        // This entry is populated by the email from the user's credential
+        case "credentialEmail":
+          entries[name] = { type: "string", value: requesterEmail };
+          break;
+        // This entry is populated by the Semaphore ID from the user's
+        // credential
+        case "credentialSemaphoreID":
+          entries[name] = {
+            type: "cryptographic",
+            value: BigInt(requesterSemaphoreId)
+          };
+          break;
+        // This entry is populated by some value from the CSV input.
+        case "input":
+          const columnName = entryConfig.source.name;
+          // Check if the configured column exists.
+          if (!(columnName in inputRow)) {
+            throw new Error(
+              `Could not find column "${entryConfig.source.name}" in CSV data, required by output field ${name}`
+            );
+          }
```

Comment:
nit: if it's an easy add, could we make this error (1) show up in the logs of the CSVPipeline and (2) set the status of the pipeline to "Error"?

in my testing, I changed the CSV column name of "zupass_description" to a value that didn't exist on the CSV input

<img width="195" alt="Screenshot 2024-06-26 at 6 34 34 PM" src="https://github.com/proofcarryingdata/zupass/assets/36896271/5cd1f789-3d8d-4d1f-b2a3-d334cd8df204">

the pipeline is basically in an invalid state now - my POD was not reissued (which makes sense, as the row is invalid)

got this error under passport-server logs but nothing else in the logs or pipeline page that indicated there was an issue:

<img width="1246" alt="Screenshot 2024-06-26 at 6 31 44 PM" src="https://github.com/proofcarryingdata/zupass/assets/36896271/de0572a7-eb9e-437c-a313-b530faee374c">

Reply:
a fix that would catch this error earlier is running this validation logic when the PipelineDefinition is being uploaded, so we never end up saving an invalid config